src/status/chat.nim

```diff
@@ -73,550 +62,697 @@ type
     notificationTypes*: seq[ActivityCenterNotificationType]
 
 include chat/utils
+include chat/async_tasks
+
+QtObject:
```

Comment:
I am not totally sure here but I don't think src/status is supposed to be couple with QT. I might be wrong but I think it should only be from src/app

Reply:
@alaibe I must say that I cannot agree on this with you. :) 
As you noticed nim source code is divided in two parts "app" which should be used for view stuff and "status" where should be placed app's logic. We're developing a Qt app, that means we're using Qt in both parts, in view part (where usually models like QAbstractListModel and other should be used) and logic part where we must inherit QObject if we want to work with signal/slot mechanism. So it's necessary on both sides.

Reply:
As I said, I wasn't totally sure it was more of an impression from looking at the code itself. Maybe @jrainville @richard-ramos  could help here?